`panic: makeslice: cap out of range` when decoding large list in const vector path

## `panic: makeslice: cap out of range` when decoding large list in const vector path

### Environment

- **nebula-go version**: v5.2.1
- **NebulaGraph server version**: Enterprise 5.2.0
- **Go version**: go1.24
- **OS**: Linux (amd64)

### Description

When a GQL query returns a single-row result containing a large list (e.g., from `collect()`), the SDK panics with `makeslice: cap out of range` during result decoding.

The query executes successfully on the server side, but the SDK crashes when trying to decode the response.

### Root Cause

In `internal/decode/resultTable_helper.go`, the `decodeAnyCompositeValue` function uses `bytesToInt16` (signed int16, max 32767) to read the list size field:

```go
// resultTable_helper.go:824-833
case types.ColumnTypeList:
    typeBytes := r.readN(1)
    sizeBytes := r.readN(2)
    // ...
    size := int(bytesToInt16(sizeBytes))    // ← signed int16, max 32767
    l := make([]*NebulaValue, 0, size)      // ← panics when size is negative
```

When the list has more than 32767 elements, the signed int16 overflows to a negative value, and `make()` panics because it receives a negative capacity.

Note that the **flat vector path** (`decodeListValue`) correctly uses `bytesToUint32`:

```go
// resultTable_helper.go:249-250
offset := bytesToUint32(header[:4])
size := bytesToUint32(header[4:8])    // ← uint32, no overflow issue
```

The inconsistency between the two paths is the core issue. This same `bytesToInt16` pattern also affects `ColumnTypeRecord`, `ColumnTypePath`, `ColumnTypeVector`, and `ColumnTypeString` decoding within `decodeAnyCompositeValue`.

### When the const path is triggered

When a query returns only **one row** (e.g., `RETURN collect(...)`), the server encodes the column as a **const vector** instead of a flat vector. The const vector decoding goes through `decodeConstValue` → `decodeAnyCompositeValue`, which is the buggy code path.

### Steps to Reproduce

```go
package main

import (
    "fmt"
    nebula "github.com/vesoft-inc/nebula-go/v5"
)

func main() {
    client, _ := nebula.NewNebulaClient("host:9669", "user", "password")
    defer client.Close()

    // Any query that returns a single-row list with >32767 elements will trigger the panic.
    // For example, a graph traversal with collect():
    resp, _ := client.Execute(`
        MATCH (n)-[e]->(m)
        WHERE n.id = "some_id"
        RETURN collect(m.id)
    `)

    // Panic occurs here during decoding, if the collected list has >32767 elements
    for resp.HasNext() {
        row, _ := resp.Next()   // panic: runtime error: makeslice: cap out of range
        _ = row
    }
}
```

### Stack Trace

```
panic: runtime error: makeslice: cap out of range

goroutine 1 [running]:
github.com/vesoft-inc/nebula-go/v5/internal/decode.decodeAnyCompositeValue(...)
        .../internal/decode/resultTable_helper.go:833 +0x6e6
github.com/vesoft-inc/nebula-go/v5/internal/decode.(*vectorDecoder).decodeConstValue(...)
        .../internal/decode/resultTable_helper.go:142 +0xb7
github.com/vesoft-inc/nebula-go/v5/internal/decode.(*vectorDecoder).decodeValue(...)
        .../internal/decode/resultTable_helper.go:115 +0x13c
github.com/vesoft-inc/nebula-go/v5/internal/decode.(*vectorWrapper).decodeValue(...)
        .../internal/decode/resultTable.go:387 +0xe9
github.com/vesoft-inc/nebula-go/v5/internal/decode.(*batch).getRowByIndex(...)
        .../internal/decode/resultTable.go:335 +0xf9
github.com/vesoft-inc/nebula-go/v5/internal/decode.(*ResultTable).Next(...)
        .../internal/decode/resultTable.go:310 +0x12a
github.com/vesoft-inc/nebula-go/v5.(*resultSet).Next(...)
        .../resultset.go:50 +0x45
```

### Suggested Fix

At minimum, change `bytesToInt16` to `bytesToUint16` in `decodeAnyCompositeValue` for the size fields of `ColumnTypeList`, `ColumnTypeRecord`, `ColumnTypePath`, `ColumnTypeVector`, and `ColumnTypeString`. This would raise the limit from 32767 to 65535.

If the server's wire protocol actually encodes these sizes as uint16, this is a straightforward SDK bug. If larger lists (>65535) need to be supported, the const encoding format itself may need to be revised to use 4 bytes, consistent with the flat encoding path.

--- a/internal/decode/resultTable_helper.go
+++ b/internal/decode/resultTable_helper.go
@@ -3,9 +3,9 @@
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
 // You may obtain a copy of the License at
-// 
-//     http://www.apache.org/licenses/LICENSE-2.0
-// 
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
 // Unless required by applicable law or agreed to in writing, software
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
@@ -808,7 +808,7 @@ func decodeAnyCompositeValue(dctx *decodeContext, value *NebulaValue, r *bytesRe
 		if r.error() != nil {
 			return r.error()
 		}
-		size := int(bytesToInt16(sizeBytes))
+		size := int(bytesToUint16(sizeBytes))
 		bs := r.readN(size)
 		if r.error() != nil {
 			return r.error()
@@ -829,7 +829,7 @@ func decodeAnyCompositeValue(dctx *decodeContext, value *NebulaValue, r *bytesRe
 		if !ok {
 			return errors.Wrap(errInvalidColumnType, "")
 		}
-		size := int(bytesToInt16(sizeBytes))
+		size := int(bytesToUint16(sizeBytes))
 		l := make([]*NebulaValue, 0, size)
 		var bitSize int
 		if size%8 != 0 {
@@ -860,7 +860,7 @@ func decodeAnyCompositeValue(dctx *decodeContext, value *NebulaValue, r *bytesRe
 		}
 	case types.ColumnTypeRecord:
 		sizeBytes := r.readN(2)
-		size := int(bytesToInt16(sizeBytes))
+		size := int(bytesToUint16(sizeBytes))
 		m := make(map[string]*NebulaValue, 0)
 		for i := 0; i < size; i++ {
 			bs := r.readN(2)
@@ -896,7 +896,7 @@ func decodeAnyCompositeValue(dctx *decodeContext, value *NebulaValue, r *bytesRe
 		nodeTypeId := int32(nodeId >> 48)
 		graphId := bytesToInt32(graphIdBytes)
 		_ = graphId
-		propSize := bytesToInt16(propSizeBytes)
+		propSize := bytesToUint16(propSizeBytes)
 		keys := make([]string, 0, propSize)
 		values := make([]*NebulaValue, 0, propSize)
 		for i := 0; i < int(propSize); i++ {
@@ -946,7 +946,7 @@ func decodeAnyCompositeValue(dctx *decodeContext, value *NebulaValue, r *bytesRe
 		graphId := bytesToInt32(graphIdBytes)
 		edgeTypeID := bytesToInt32(edgeTypeIdBytes)
 		edgeRank := bytesToInt64(edgeRankBytes)
-		propSize := bytesToInt16(propSizeBytes)
+		propSize := bytesToUint16(propSizeBytes)
 		propNames := make([]string, 0, propSize)
 		propValues := make([]*NebulaValue, 0, propSize)
 		noDirectType := edgeTypeID & 0x3FFFFFFF

